adapter/pkg/synchronizer/apis_fetcher.go

```diff
@@ -155,70 +212,34 @@ func FetchAPIs(id *string, gwLabel []string, c chan SyncAPIResponse, serviceURL
 		q.Add(gwType, envoy)
 	}
 	req.URL.RawQuery = q.Encode()
+
 	// Setting authorization header
+	basicAuth := "Basic " + auth.GetBasicAuth(userName, password)
 	req.Header.Set(Authorization, basicAuth)
 	// If API UUID list is present, set the content-type header
 	if apiUUIDList != nil {
 		req.Header.Set("Content-Type", "application/json")
 	}
-	// Make the request
-	logger.LoggerSync.Debug("Sending the controle plane request")
-	resp, err = client.Do(req)
-	// In the event of a connection error, the error would not be nil, then return the error
-	// If the error is not null, proceed
-	if err != nil {
-		logger.LoggerSync.Errorf("Error occurred while retrieving APIs from API manager: %v", err)
-		respSyncAPI.Err = err
-		respSyncAPI.Resp = nil
-		c <- respSyncAPI
-		return
-	}
+	return req
+}
 
-	// get the response in the form of a byte slice
-	respBytes, err = ioutil.ReadAll(resp.Body)
+// RetryFetchingAPIs function keeps retrying to fetch APIs from runtime-artifact endpoint.
+func RetryFetchingAPIs(c chan SyncAPIResponse, data SyncAPIResponse, endpoint string, sendType bool,
+	queryParamMap map[string]string) {
+	retryInterval := workerPool.controlPlaneParams.retryInterval
 
-	// If the reading response gives an error
-	if err != nil {
-		logger.LoggerSync.Errorf("Error occurred while reading the response: %v", err)
-		respSyncAPI.Err = err
-		respSyncAPI.ErrorCode = resp.StatusCode
-		respSyncAPI.Resp = nil
-		c <- respSyncAPI
-		return
-	}
-	// For successful response, return the byte slice and nil as error
-	if resp.StatusCode == http.StatusOK {
-		respSyncAPI.Err = nil
-		respSyncAPI.Resp = respBytes
-		c <- respSyncAPI
-		return
+	// Retry fetching from control plane after a configured time interval
+	if retryInterval == 0 {
+		// Assign default retry interval
+		retryInterval = 5
 	}
-	// If the response is not successful, create a new error with the response and log it and return
-	// Ex: for 401 scenarios, 403 scenarios.
-	logger.LoggerSync.Errorf("Failure response: %v", string(respBytes))
-	respSyncAPI.Err = errors.New(string(respBytes))
-	respSyncAPI.Resp = nil
-	respSyncAPI.ErrorCode = resp.StatusCode
-	c <- respSyncAPI
-	return
-}
+	logger.LoggerSync.Debugf("Time Duration for retrying: %v", retryInterval*time.Second)
+	time.Sleep(retryInterval * time.Second)
+	logger.LoggerSync.Infof("Retrying to fetch API data from control plane for the API %q.", data.APIUUID)
+	channelFillPercentage := float64(len(workerPool.internalQueue)) / float64(cap(workerPool.internalQueue)) * 100
+	logger.LoggerSync.Infof("Workerpool channel size as a percentage is : %f", channelFillPercentage)
+	FetchAPIs(&data.APIUUID, data.GatewayLabels, c, endpoint, sendType, nil, queryParamMap)
 
```

Comment:
Do we have a log indicating that the retry attempt successful?

Reply:
https://github.com/wso2/product-microgateway/blob/45532a6055d0588e4a774430ab84b32753166ecd/adapter/internal/synchronizer/apis_fetcher.go#L175
We improved the log such that APIID is included too.

Reply:
Ok got it.. 
Thanks..